### PR TITLE
WHO 9 cards (Doctors)

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/the_eighth_doctor.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Name:The Eighth Doctor
+ManaCost:4 W U
+Types:Legendary Creature Time Lord Doctor
+PT:4/4
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigMill | TriggerDescription$ When CARDNAME enters the battlefield, mill three cards, then put a +1/+1 counter on CARDNAME for each creature card in your graveyard. (To mill a card, put the top card of your library into your graveyard.)
+SVar:TrigMill:DB$ Mill | Defined$ You | NumCards$ 3
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_eighth_doctor.txt:6]: incomplete

File: forge-gui/res/cardsfolder/upcoming/the_eighth_doctor.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Name:The Eighth Doctor
+ManaCost:4 W U
+Types:Legendary Creature Time Lord Doctor
+PT:4/4
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigMill | TriggerDescription$ When CARDNAME enters the battlefield, mill three cards, then put a +1/+1 counter on CARDNAME for each creature card in your graveyard. (To mill a card, put the top card of your library into your graveyard.)
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_eighth_doctor.txt:5]: wrong Description

File: forge-gui/res/cardsfolder/upcoming/the_eighth_doctor.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+T:Mode$ SpellCast | ValidCard$ Card | ValidSA$ Spell.MayPlaySource | Static$ True | Execute$ DBAnimate | TriggerZones$ Battlefield | Secondary$ True | TriggerDescription$ Once during each of your turns, you may play a historic land or cast a historic permanent spell from your graveyard. If you do, it gains "If this permanent would leave the battlefield, exile it instead of putting it anywhere else."
+T:Mode$ LandPlayed | ValidCard$ Land | ValidSA$ SpellAbility.MayPlaySource | Static$ True | Execute$ DBAnimate | TriggerZones$ Battlefield | Secondary$ True | TriggerDescription$Once during each of your turns, you may play a historic land or cast a historic permanent spell from your graveyard. If you do, it gains "If this permanent would leave the battlefield, exile it instead of putting it anywhere else."
+SVar:DBAnimate:DB$ Animate | Defined$ TriggeredCard | Duration$ Permanent | Triggers$ TrigDieExile
+SVar:TrigDieExile:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Any | ValidCard$ Card.Self | Execute$ TrigExile | TriggerDescription$ If this permanent would leave the battlefield, exile it instead of putting it anywhere else.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_eighth_doctor.txt:11]: this is not a trigger

File: forge-gui/res/cardsfolder/upcoming/the_eleventh_doctor.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+Name:The Eleventh Doctor
+ManaCost:1 W U
+Types:Legendary Creature Time Lord Doctor
+PT:3/2
+T:Mode$ DamageDone | ValidSource$ Card.Self | Execute$ TrigChangeZone | CombatDamage$ True | ValidTarget$ Player | TriggerZones$ Battlefield | TriggerDescription$ I. AM. TALKING! — Whenever CARDNAME deals combat damage to a player, you may exile a card from your hand with a number of time counters on it equal to its mana value. If it doesn't have suspend, it gains suspend.
+SVar:TrigChangeZone:DB$ ChangeZone | Origin$ Hand | Destination$ Exile | ChangeType$ Card.nonLand | WithCountersType$ TIME | WithCountersAmount$ X | RememberChanged$ True | SubAbility$ GiveSuspend
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_eleventh_doctor.txt:6]: wrong type

File: forge-gui/res/cardsfolder/upcoming/the_fifth_doctor.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+Name:The Fifth Doctor
+ManaCost:2 W U
+Types:Legendary Creature Time Lord Doctor
+PT:2/2
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigCounters | TriggerDescription$ Peaceful Coexistence — At the beginning of your end step, put a +1/+1 counter on each creature you control that didn't attack or enter the battlefield this turn. Untap those creatures.
+SVar:TrigCounters:DB$ PutCounterAll | ValidCards$ Creature.YouCtrl+!ThisTurnEntered+!attackedThisTurn | CounterType$ P1P1 | CounterNum$ 1 | RememberPut$ True | SubAbility$ DBUntap
+SVar:DBUntap:DB$ UntapAll | ValidCards$ Creature.IsRemembered | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_fifth_doctor.txt:7]: don't use remembered

File: forge-gui/res/cardsfolder/upcoming/the_first_doctor.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+SVar:TrigSearch:DB$ ChangeZone | Hidden$ True | Origin$ Library | OriginAlternative$ Graveyard | AlternativeMessage$ Would you like to search your library with this ability? If you do, your library will be shuffled. | Destination$ Hand | ChangeType$ Card.YouOwn+namedTARDIS
+T:Mode$ SpellCast | ValidCard$ Card.withCascade | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigPutCounter | TriggerDescription$ Whenever you cast a spell with cascade, put a +1/+1 counter on target artifact or creature
+SVar:TrigPutCounter:DB$ PutCounter | ValidTgts$ Creature,Artifact | TgtPrompt$ Select target artifact or creature | CounterType$ P1P1 | CounterNum$ 1 
+DeckNeeds:Name$TARDIS
+DeckHas:Ability$Counters
+Oracle:When The First Doctor enters the battlefield, search your library and/or graveyard for a card named TARDIS, reveal it, and put it into your hand. If you search your library this way, shuffle.\nWhenever you cast a spell with cascade, put a +1/+1 counter on target artifact or creature
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_first_doctor.txt:11]: .

File: forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt [hunk 7]
```diff
@@ -0,0 +1,15 @@
+Name:The Seventh Doctor
+ManaCost:3 W U
+Types:Legendary Creature Time Lord Doctor
+PT:3/6
+T:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigGuess | TriggerZones$ Battlefield | TriggerDescription$ Whenever CARDNAME deals combat damage to a player, choose a card in your hand. That player guesses whether the card's mana value is greater than 4. If the player guessed wrong, you may cast the card without paying its mana cost.
+SVar:TrigGuess:DB$ ChooseCard | ChoiceZone$ Hand | Defined$ You | Choices$ Card.YouOwn | AILogic$ RandomNonLand | SubAbility$ DBGuess
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt:6]: Mandatory

File: forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt [hunk 8]
```diff
@@ -0,0 +1,15 @@
+Name:The Seventh Doctor
+ManaCost:3 W U
+Types:Legendary Creature Time Lord Doctor
+PT:3/6
+T:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigGuess | TriggerZones$ Battlefield | TriggerDescription$ Whenever CARDNAME deals combat damage to a player, choose a card in your hand. That player guesses whether the card's mana value is greater than 4. If the player guessed wrong, you may cast the card without paying its mana cost.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt:5]: wrong description

File: forge-gui/res/cardsfolder/upcoming/the_sixth_doctor.txt [hunk 9]
```diff
@@ -0,0 +1,9 @@
+Name:The Sixth Doctor
+ManaCost:4 G U
+Types:Legendary Creature Time Lord Doctor
+PT:3/3
+T:Mode$ SpellCast | TriggerZones$ Battlefield | ValidCard$ Card.Historic | ValidActivatingPlayer$ You | ActivationLimit$ 1 | Execute$ TrigCopy | TriggerDescription$ Time Lord's Prerogative — Whenever you cast a historic spell, copy it, except the copy isn't legendary. This ability triggers only once each turn. (Artifacts, legendaries, and Sagas are historic. A copy of a permanent spell becomes a token.)
+SVar:TrigCopy:DB$ CopySpellAbility | Defined$ TriggeredSpellAbility | NonLegendary$ True | MayChooseTarget$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_sixth_doctor.txt:6]: no MayChoose

File: forge-gui/res/cardsfolder/upcoming/the_fifth_doctor.txt [hunk 10]
```diff
@@ -0,0 +1,10 @@
+Name:The Fifth Doctor
+ManaCost:2 W U
+Types:Legendary Creature Time Lord Doctor
+PT:2/2
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigCounters | TriggerDescription$ Peaceful Coexistence — At the beginning of your end step, put a +1/+1 counter on each creature you control that didn't attack or enter the battlefield this turn. Untap those creatures.
+SVar:TrigCounters:DB$ PutCounterAll | ValidCards$ Creature.YouCtrl+!ThisTurnEntered+!attackedThisTurn | CounterType$ P1P1 | CounterNum$ 1 | SubAbility$ DBUntap
+SVar:DBUntap:DB$ UntapAll | ValidCards$ Creature.YouCtrl+!ThisTurnEntered+!attackedThisTurn
+DeckHas:Ability$Counter
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_fifth_doctor.txt:8]: typos

File: forge-gui/res/cardsfolder/upcoming/the_ninth_doctor.txt [hunk 11]
```diff
@@ -0,0 +1,8 @@
+Name:The Ninth Doctor
+ManaCost:1 U R
+Types:Legendary Creature Time Lord Doctor
+PT:2/4
+K:Haste
+T:Mode$ Untaps | ConditionPhases$ Untap | ValidCard$ Card.Self | TriggerZones$ Battlefield | Execute$ TrigAddPhase | TriggerDescription$ Into the TARDIS — Whenever CARDNAME becomes untapped during your untap step, you get an additional upkeep step after this step.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_ninth_doctor.txt:6]: the timing param you used does not exist 😢

File: forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt [hunk 12]
```diff
@@ -0,0 +1,15 @@
+Name:The Seventh Doctor
+ManaCost:3 W U
+Types:Legendary Creature Time Lord Doctor
+PT:3/6
+T:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigGuess | TriggerZones$ Battlefield | TriggerDescription$ When CARDNAME attacks, choose a card in your hand. Defending player guesses whether that card's mana value is greater than the number of artifacts you control. If they guessed wrong, you may cast it without paying its mana cost. If you don't cast a spell this way, investigate. (Create a Clue token. It's an artifact with "{2}, Sacrifice this artifact: Draw a card.")
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt:5]: wrong trigger

File: forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+SVar:TrigGuess:DB$ ChooseCard | ChoiceZone$ Hand | Mandatory$ True | Defined$ You | Choices$ Card.YouOwn | AILogic$ RandomNonLand | SubAbility$ DBGuess
+SVar:DBGuess:DB$ GenericChoice | Defined$ TriggeredTarget | Choices$ GuessGreaterArtifacts,GuessNotGreaterArtifacts | AILogic$ Random | ShowChoice$ True
+SVar:GuessGreaterArtifacts:DB$ Play | Controller$ You | Defined$ ChosenCard | Optional$ True | WithoutManaCost$ True | ConditionDefined$ ChosenCard | ConditionPresent$ Card.cmcLEX+nonLand | ConditionCompare$ GE1 | SubAbility$ DBCleanup | SpellDescription$ That card's mana value is greater the number of artifact attacking player controls
+SVar:GuessNotGreaterArtifacts:DB$ Play | Controller$ You | Defined$ ChosenCard | Optional$ True | WithoutManaCost$ True | ValidSA$ Spell | ConditionDefined$ ChosenCard | ConditionPresent$ Card.cmcGTX+nonLand | SubAbility$ DBCleanup | ConditionCompare$ GE1 | SpellDescription$ That card's mana value is not greater than the number of artifact attacking player controls
+SVar:DBInvestigate:DB$ Investigate | ConditionDefined$ Remembered | ConditionPresent$ Card | ConditionCompare$ EQ0 | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt:10]: nothing gets remembered...

File: forge-gui/res/cardsfolder/upcoming/the_second_doctor.txt [hunk 14]
```diff
@@ -0,0 +1,12 @@
+Name:The Second Doctor
+ManaCost:2 W U
+Types:Legendary Creature Time Lord Doctor
+PT:2/4
+S:Mode$ Continuous | Affected$ Player | SetMaxHandSize$ Unlimited | Description$ Players have no maximum hand size.
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigChoose | TriggerDescription$ How Civil of You — At the beginning of your end step, each player may draw a card. Each opponent who does can't attack you or permanents you control during their next turn.
+SVar:TrigChoose:DB$ RepeatEach | RepeatPlayers$ Player | RepeatSubAbility$ DBAllowDraw
+SVar:DBAllowDraw:DB$ Draw | OptionalDecider$ True | Defined$ Remembered | RememberDrawn$ True | NumCards$ 1 | SubAbility$ DBEffect 
+SVar:DBEffect:DB$ Effect | RememberObjects$ Remembered | ConditionDefined$ Remembered | ConditionPresent$ Card | StaticAbilities$ STCantAttack | Duration$ UntilYourNextTurn | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_second_doctor.txt:9]: probably easiest if you give the effect to the opponents so the Duration can be adjusted seamlessly

File: forge-gui/res/cardsfolder/upcoming/the_second_doctor.txt [hunk 15]
```diff
@@ -0,0 +1,12 @@
+Name:The Second Doctor
+ManaCost:2 W U
+Types:Legendary Creature Time Lord Doctor
+PT:2/4
+S:Mode$ Continuous | Affected$ Player | SetMaxHandSize$ Unlimited | Description$ Players have no maximum hand size.
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigChoose | TriggerDescription$ How Civil of You — At the beginning of your end step, each player may draw a card. Each opponent who does can't attack you or permanents you control during their next turn.
+SVar:TrigChoose:DB$ RepeatEach | RepeatPlayers$ Player | RepeatSubAbility$ DBAllowDraw
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_second_doctor.txt:7]: shouldn't need Repeat at all
just let all draw

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/the_second_doctor.txt:7]: If I understand this card correctly every player can choose for itself if they want to draw ?

File: forge-gui/res/cardsfolder/upcoming/the_ninth_doctor.txt (hunk 11 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_ninth_doctor.txt:6]: no, for which part do you think?

File: forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt (hunk 13 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt]: you got rid of the cleanup, but the problem is you should remember something

File: forge-gui/res/cardsfolder/upcoming/the_second_doctor.txt (hunk 15 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_second_doctor.txt:7]: sure

File: forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt (hunk 12 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt]: when you change a trigger you need to be mindful to use the right triggeringObjects too

File: forge-gui/res/cardsfolder/upcoming/the_ninth_doctor.txt (hunk 11 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/the_ninth_doctor.txt:6]: `ActivationPhases$ ` and `UntapStep$ True ` don't seem to work as well. What's the param I'm missing ? 🤔

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_ninth_doctor.txt:6]: Sorry to say it's just `Phase$`
For the second aspect I'd recommend just adding `+ActivePlayerCtrl` to the valid

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/the_ninth_doctor.txt:6]: Thanks, will do

File: forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt (hunk 12 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt]: `TriggeredDefender `is not correct ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt]: no, use DefendingPlayer since it could also attack a Card and then you're in trouble

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_seventh_doctor.txt]: it's still a triggeringObject

**Review by tool4ever — APPROVED** [review]

**Pull request merged**